tmt/steps/report/__init__.py

```diff
@@ -54,8 +55,13 @@ def go(self) -> None:
             return
 
         # Perform the reporting
-        for phase in self.phases():
-            phase.go()
+        for phase in self.phases(classes=(Action, ReportPlugin)):
+            # TODO: I don't understand this, but mypy seems to be confused about the type
+            # of `phase`. Mypy in my Code reports correct `Action | ReportPlugin` union,
+            # but pre-commit's mypy sees `Phase` - which should not be the right answer
+            # since `classes` is clearly not `None`. Adding `cast()` to overcome this
+            # because I can't find the actual error :/
+            cast(Union[Action, ReportPlugin], phase).go()
```

Comment:
@happz last week I added a task to revisit all casts, so we should get back to this.

Reply:
Shall I take any action in this PR?

Reply:
nope, just commented FYI, sorry, should have mentioned.